backend/game/cards/discard_card.py

```diff
@@ -4,6 +4,5 @@
 
 
 class DiscardCard(Card):
-    @abstractmethod
     def immediate_effect(self, player_that_bought_the_card, other_players):
         raise NotImplementedError
```

Comment:
This whole function can be removed as it is no longer needed here, it may also be possible to roll up all this discard/keep card modules into the Cards module.